### ISSUE TITLE
uuid generation requires external library

### ISSUE THREAD

**Issue by rtyler** [freebsd/openlaunchd#2]
Looks like Apple OSes uses a bundled version of [e2fsprogs' libuuid](http://git.kernel.org/cgit/fs/ext2/e2fsprogs.git/tree/lib/uuid) to provide `<uuid/uuid.h>` 

```
cc  -O2 -pipe  -Werror -g -I/usr/home/tyler/source/github/openlaunchd/liblaunch/../src -std=gnu99 -Qunused-arguments -fstack-protector  -c liblaunch.c -o liblaunch.o
In file included from liblaunch.c:23:
./launch_priv.h:28:10: error: 'launch.h' file not found with <angled> include; use "quotes" instead
#include <launch.h>
         ^~~~~~~~~~
         "launch.h"
./launch_priv.h:31:10: fatal error: 'uuid/uuid.h' file not found
#include <uuid/uuid.h>
         ^
2 errors generated.
*** Error code 1

Stop.
make[1]: stopped in /usr/home/tyler/source/github/openlaunchd/liblaunch
*** Error code 1

Stop.
make: stopped in /usr/home/tyler/source/github/openlaunchd
```

This should either be bundled into the openlaunchd tree, or enumerated as a dependency.

**Comment by CameronNemo** [comment]
e2fsprogs is only in FreeBSD ports, it is at version 1.34, and was last modified in 2007. Upstream is at version 1.42, and that was released six months ago. We should find out if 1.34 is enough for liblaunch, but pretty much all of the changes in e2fsprogs are bug fixes, so 1.42 should be used regardless.

**Comment by rtyler** [comment]
@CameronNemo Where are you getting the 1.34 version? The version of e2fsprogs that's in [head of the ports tree](https://svn0.us-west.freebsd.org/ports/head/sysutils/e2fsprogs/Makefile) lists the port version at 1.42.8

I'm fairly certain there will be some libuuid compatibility issues since Apple was using a forked version of e2fsprogs, but on its face the libuuid from my Apple machine and my FreeBSD machine look close enough alike
